### PR TITLE
feat: useNotification unit test

### PR DESCRIPTION
**Pull request by maxiroellplenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/composables/useNotification/__tests__/useNotification.spec.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+            dismissTimeout: 10
+        });
+
+        expect(data.value.length).toBe(1);
+
+        await new Promise((resolve) => setTimeout(resolve, 10));
```
Comment (ksted) [inline apps/web/composables/useNotification/__tests__/useNotification.spec.ts]: **suggestion**: Add a threshold test. That is, set a timeout for 9 seconds and check if `length` is still `1`.

Comment (maxiroellplenty) [reply apps/web/composables/useNotification/__tests__/useNotification.spec.ts]: done, it now tests the default timeout of 3000ms and checks right before the end if the message is still there

File: apps/web/composables/useNotification/__tests__/useNotification.spec.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        send({
+            message: 'Test alert error with a longer message',
+            type: 'negative',
+            action: {
+                text: 'action',
+                onClick: () => {},
+            },
+        });
+
+        expect(data.value.length).toBe(1);
```
Comment (ksted) [inline apps/web/composables/useNotification/__tests__/useNotification.spec.ts]: **suggestion**: Add a second message to the queue to prove that this is different from dismissing a single message.

File: apps/web/composables/useNotification/__tests__/useNotification.spec.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+        clear();
+
+        expect(data.value.length).toBe(0);
+    });
+
+    it('should remove the oldest notification when the maximum number of notifications is reached', async () => {
```
Comment (ksted) [inline apps/web/composables/useNotification/__tests__/useNotification.spec.ts:63]: **suggestion**: The test case currently only checks that a message gets removed, not which one. Maybe `i` could be part of the message and have an assertion to that effect.

Comment (maxiroellplenty) [reply apps/web/composables/useNotification/__tests__/useNotification.spec.ts:63]: good idea, done 👍

File: apps/web/composables/useNotification/__tests__/useNotification.spec.ts (hunk 2 above)
Comment (maxiroellplenty) [reply apps/web/composables/useNotification/__tests__/useNotification.spec.ts]: done 👍

**Review by ksted — APPROVED** [review]

**Pull request merged**